### DISCUSSION TITLE
Opacity for the background of the text layer or text in a different color

### DISCUSSION THREAD

**Discussion by tellts** [pymupdf/PyMuPDF#2665] [Q&A]
Hello. Can you please tell me how I can make the color of a text layer red? And maybe increase it a little. And it's possible to add an opaque background to just the text, not the entire page. I translated the PDF through Google Translator, which had the letters under the page images and now the black letters of the original are superimposed on a black text layer in a different language. Because of this, it is not clear what is written. I need to either change the color of the text layer or make an opaque background under the lines of text.

I tried the following:
```
import fitz

# Open the input PDF file
doc = fitz.open("input.pdf")

# Search for the word "study" in the text layer of the first page
page = doc[0]
highlight = "study"
rl = page.search_for(highlight, quads=True)  # Search for the word and get its quads
if not rl:
    print("The word 'study' was not found in the PDF file.")
else:
    annot = page.add_highlight_annot(rl[0])  # Add a highlight annotation using the first quad
    print(annot)  # Print some information about the annotation

    # Save the modified PDF under a new name
    doc.save("new_input.pdf")
doc.close()
```

I also did two other examples. And the background always turned out transparent.

**Comment by JorjMcKie** [comment]
I am not 100% sure I understand.
The text to highlight is underneath an image?

Apart from this: you can play with the hgighlight annotation's blend mode. The standard is "Multiply" which makes underlying text come through clearly ... if it is darker than the highlight color.

Otherwise you cannot selectively change the color of existing text ... you have to rewrite it.

**Comment by tellts** [comment]
Black text in another language (not English, as this is a translation) is on top of the image. The image shows black text in English. I need the English text not to interfere with reading the translation.

**Reply by JorjMcKie** [reply]
What you can do is removing unwanted text (using redaction annotations). When adding new (translated) text, you have a number of choices how you can do this. You can certainly choose the color, but also set its transparency.

**Reply by tellts** [reply]
I don’t understand what exactly to delete? The translated text is available in the PDF document. That is, you need to take a copy of it, color it, and delete the original text and put the colored text in its place?

**Reply by JorjMcKie** [reply]
To remove all existing text without touching images on the page, do this:
```python 
page.add_redact_annot(page.rect)
page.apply_redactions(images=fitz.PDF_REDACT_IMAGE_NONE)
```

**Reply by tellts** [reply]
Thank you. I'll think about your example in the coming days (I'm leaving for a while tomorrow).

**Reply by JorjMcKie** [reply]
Ok, so you have the complete text on the page: English plus its translation.
Now you want to make the English stuff invisible, right?

**Reply by tellts** [reply]
Yes, that's right. I want to make the English text invisible. The English text is not on the text layer, but on the page image.

**Reply by JorjMcKie** [reply]
So the English text is not PDF text, but part of the image.
Well in that case, you must modify the image itself. Where there is English text, you will want to appear empty rectangles of some color.
Once this is done, write the translation over those emptied areas ... about right?

**Reply by tellts** [reply]
Yes, this will solve the problem.
That is, so that the places where the translation is located are painted over.

**Comment by JorjMcKie** [comment]
Do you have the "en_rects" information?
